File: algorithm/src/main/java/baekjoon/p11652/Main.java
package baekjoon.p11652;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long[] inputs = new long[n];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextLong();
        }

        System.out.println(solve(inputs));
    }

    private static long solve(long[] inputs) {
        HashMap<Long, Integer> cnt = new HashMap<>();
        for (long e : inputs) {
            if (cnt.containsKey(e)) {
                cnt.put(e, cnt.get(e) + 1);
            } else {
                cnt.put(e, 1);
            }
        }

        List<Long> sortedKey = new ArrayList<>(cnt.keySet());
        Collections.sort(sortedKey);
        long max = sortedKey.get(0);

        for (long e : sortedKey) {
            if (cnt.get(e) > cnt.get(max)) {
                max = e;
            }
        }

        return max;
    }

}

File: algorithm/src/main/java/baekjoon/p11652/Main.kt
package baekjoon.p11652

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = List(n) { readLine()!!.toLong() }
    println(solve(inputs))
}

private fun solve(inputs: List<Long>): Long {
    val cnt = mutableMapOf<Long, Int>()
    inputs.forEach { cnt[it] = cnt.getOrDefault(it, 0).inc() }

    return cnt.maxWith(compareBy({ it.value }, { -it.key }))!!.key
}